### ISSUE TITLE
Arrays should be indexable from the end too

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#2974]
I want to be able to access the last element(s) in an array. 

KSQL support accessing elements based on offset from the start: 

```
ksql> SELECT SCHEDULE_SEGMENT_LOCATION[0] FROM SCHEDULE_01 LIMIT 1;
{"record_identity":"LO","engineering_allowance":"3","line":"AL","public_departure":"0506","tiploc_code":"YORK","departure":"0508","location_type":"LO","platform":"9"}
Limit Reached
Query terminated
```

But not the end: 

```
ksql> SELECT SCHEDULE_SEGMENT_LOCATION[-1] FROM SCHEDULE_01 LIMIT 1;
null
Limit Reached
Query terminated
```
